androidHyperskillApp/src/main/java/org/hyperskill/app/android/study_plan/adapter/StudyPlanActivityAdapterDelegate.kt

```diff
@@ -56,13 +56,21 @@ class StudyPlanActivityAdapterDelegate(
                 }
             }
 
+            with(binding.activityBadges) {
+                isVisible = data.isIdeRequired || isProgressVisible
+            }
```

Comment:
You should not always use `with` scope function. It makes sense in case you have more than 1 call to the subject.